CircleSpace/CircleSpaceServiceLib/Service/SqlCricleSpaceService.cs

```diff
@@ -0,0 +1,344 @@
+                {
+                    PageRoute = model.Route,
+                    ID = db.Pages.Max(p => p.ID) + 1,
+                    OwnerID = model.OwnerID,
+                    Header = model.Header
+
+                };
+            }
+        }
+        public void DeleteContributorFromPage(PageModel page, UserModel contributor)
+        {
+            using (var db = new CircleSpaceEntities())
+            {
+
+                var query = db.Pages.Where(p => p.ID == page.ID).First();
+                query.Users.Remove(db.Users.Where(u => u.ID == contributor.ID).First());
+                db.SaveChanges();
+
+            }
+        }
+
+        public void DeletePage(PageModel model)
+        {
+            using (var db = new CircleSpaceEntities())
+            {
+                var pageToDelete = db.Pages.Where(p => p.ID == model.ID).First();
+                db.Pages.Remove(pageToDelete);
+                db.SaveChanges();
+            }
+        }
+
+        public List<PageModel> GetContributorPages(UserModel model)
+        {
+            List<PageModel> list = new List<PageModel>();
+            using (var db = new CircleSpaceEntities())
+            {
+                var user = db.Users.Where(u => u.ID == model.ID).First();
+                var pages = user.Pages1.ToList();
+                pages.ForEach(page =>
+                {
+                    PageModel p = PageToPageModel(page);
+                    list.Add(p);
+                });
+            }
+
+            return list;
+        }
+
+        public List<LayoutModel> GetLayouts()
+        {
+            List<LayoutModel> list = new List<LayoutModel>();
+            using (var db = new CircleSpaceEntities())
+            {
+                var query = db.Layouts.Select(x => x);
+                list = LayoutsToListOfLayouts(query.ToList());
+            }
+
+            return list;
+        }
+
+        public LayoutModel GetLayoutWithID(int id)
+        {
+            LayoutModel l = new LayoutModel();
+            using (var db = new CircleSpaceEntities())
+            {
+                l = LayoutToLayoutModel(db.Layouts.Where(x => x.ID == id).First());
+            }
+            return l;
+        }
+
+        public LayoutModel GetLayoutWithOwner(UserModel model)
+        {
+            LayoutModel l = new LayoutModel();
+            using (var db = new CircleSpaceEntities())
+            {
+                l = LayoutToLayoutModel(db.Layouts.Where(x => x.OwnerID == model.ID).First());
+            }
+            return l;
+        }
+
+        public LayoutModel GetLayoutWithTag(params string[] tags)
+        {
+            //LayoutModel l = new LayoutModel();
+            //using (var db = new CircleSpaceEntities())
+            //{
+            throw new NotImplementedException();
+
+            //}
+        }
+
+        public LayoutModel GetLayoutWithType(LayoutTypes type)
+        {
+            LayoutModel l = new LayoutModel();
```

Comment:
This method should get all Layouts of a certain type and not just the **first one it finds**.